Software/src/LedDeviceWarls.cpp

```diff
@@ -56,22 +54,30 @@ void LedDeviceWarls::setColors(const QList<QRgb> & colors)
 
 		for (int i = 0; i < m_colorsBuffer.count(); i++)
 		{
-			StructRgb color = m_colorsBuffer[i];
-
-			// Reduce 12-bit colour information
-			color.r = color.r >> 4;
-			color.g = color.g >> 4;
-			color.b = color.b >> 4;
-
-			m_writeBuffer.append(i);
-			m_writeBuffer.append(color.r);
-			m_writeBuffer.append(color.g);
-			m_writeBuffer.append(color.b);
+			if (colors[i] != m_colorsSaved[i])
+			{
+				StructRgb color = m_colorsBuffer[i];
+
+				// Reduce 12-bit colour information
+				color.r = color.r >> 4;
+				color.g = color.g >> 4;
+				color.b = color.b >> 4;
+
+				m_writeBuffer.append(i);
+				m_writeBuffer.append(color.r);
+				m_writeBuffer.append(color.g);
+				m_writeBuffer.append(color.b);
+			}
 		}
+
+		m_colorsSaved = colors;
 	}
 
-	bool ok = writeBuffer(m_writeBuffer);
-	emit commandCompleted(ok);
+	if (m_writeBuffer.size() != m_writeBufferHeader.size())
+	{
+		bool ok = writeBuffer(m_writeBuffer);
+		emit commandCompleted(ok);
+	}
```

Comment:
This should probably be something like
```suggestion
	bool ok = true;
	if (m_writeBuffer.size() != m_writeBufferHeader.size())
		ok = writeBuffer(m_writeBuffer);

	emit commandCompleted(ok);
```
otherwise it's treated as a fail and the tray icon goes red

also that `if ...` works because of the `255` timeout value, maybe this deserves a comment here, just in case...

Reply:
Hadn't noticed that, cheers! Not quite sure what you mean about the `255` value?

Reply:
See? That's why it needs a comment :D
This value
https://github.com/psieg/Lightpack/blob/e27426f00a0e2142578e20dee12e7d9e7bbbf8c8/Software/src/LedDeviceWarls.cpp#L143

Reply:
I know the value you mean, it just isn't obvious why the `if ...` only works because of it?

Reply:
and while we are on that subject, that value has practical implications, WLED will either:
 * go back to doing whatever it was doing before it was taken over by Prismatik
 * or it'll sit doing nothing until something else tells it otherwise (current state)

I don't know which one people expect, maybe it should be a setting ?..

Reply:
the `if ...` works because WLED won't timeout if you don't send updates (see above)

In static moodlamp mode, with a timeout of something like 1-2 seconds (as suggested by [WLED](https://github.com/Aircoookie/WLED/wiki/UDP-Realtime-Control)), it goes back to showing animations/colors it was doing before Prismatik started sending data

Reply:
Ah I see what you mean now. I wasn't sure whether WLED supports buffers with no LEDs specified, my code didn't :D